saltstack/base/salt/ccmv2-inverting-proxy-agent/cdp/bin/ccmv2/update-inverting-proxy-agent-values.sh

```diff
@@ -36,22 +36,23 @@ AGENT_CERT_PATH=$5
 TRUSTED_BACKEND_CERT_PATH=$6
 TRUSTED_PROXY_CERT_PATH=$7
 
-VALUES_FILE=/cdp/bin/ccmv2/inverting-proxy-agent-values-${BACKEND_ID}.sh
-
-cat > ${VALUES_FILE} <<EOF
-INVERTING_PROXY_URL=${INVERTING_PROXY_URL}
-BACKEND_ID=${BACKEND_ID}
-BACKEND_ADDRESS=${BACKEND_HOST}:${BACKEND_PORT}
-AGENT_KEY_PATH=${AGENT_KEY_PATH}
-AGENT_CERT_PATH=${AGENT_CERT_PATH}
-TRUSTED_BACKEND_CERT_PATH=${TRUSTED_BACKEND_CERT_PATH}
-TRUSTED_PROXY_CERT_PATH=${TRUSTED_PROXY_CERT_PATH}
-SCHEME=${SCHEME}
-HTTP_PROXY_URL=${HTTP_PROXY_URL}
+CONFIG_FILE=/cdp/bin/ccmv2/config-${BACKEND_ID}.toml
+
+cat > ${CONFIG_FILE} <<EOF
+[agent]
+proxy = "${INVERTING_PROXY_URL}"
+backendID = "${BACKEND_ID}"
+certificatePath = "${AGENT_CERT_PATH}"
+keyPath = "${AGENT_KEY_PATH}"
+trustedServerCertificatePath = "${TRUSTED_PROXY_CERT_PATH}"
+scheme = "${SCHEME}"
+host = "${BACKEND_HOST}:${BACKEND_PORT}"
+trustedBackendCertificatePath = "${TRUSTED_BACKEND_CERT_PATH}"
+httpProxyURL = "${HTTP_PROXY_URL}"
 EOF
 
-if [ -f "$VALUES_FILE" ]; then
-    chmod 740 ${VALUES_FILE}
+if [ -f "$CONFIG_FILE" ]; then
+    chmod 740 ${CONFIG_FILE}
```

Comment:
Config file does not need to be executable. Can you check if 440 is enough for this?

Reply:
You are right. Earlier it needed to be executable as we were running the values file. Now that is not required. So 640 should be good enough